lib/process-services-cloud/src/lib/process-cloud/services/process-filter-cloud.service.ts

```diff
@@ -0,0 +1,130 @@
+import { Observable } from 'rxjs';
+import { ProcessFilterRepresentationModel, ProcessQueryModel } from '../models/process-filter-cloud.model';
+
+@Injectable()
+export class ProcessFilterCloudService {
+
+    constructor(private storage: StorageService) {
+    }
+
+    /**
+     * Creates and returns the default filters for a process app.
+     * @param appName Name of the target app
+     * @returns Observable of default filters just created
+     */
+    public createDefaultFilters(appName: string): Observable<ProcessFilterRepresentationModel[]> {
+        const allProcessesFilter = this.getAllProcessesFilter(appName);
+        this.addFilter(allProcessesFilter);
+        const runningProcessesFilter = this.getRunningProcessesFilter(appName);
+        this.addFilter(runningProcessesFilter);
+        const completedProcessesFilter = this.getCompletedProcessesFilter(appName);
+        this.addFilter(completedProcessesFilter);
+
+        return this.getProcessFilters(appName);
+    }
+
+    /**
+     * Gets all process instance filters for a process app.
+     * @param appName Name of the target app
+     * @returns Observable of process filter details
+     */
+    getProcessFilters(appName: string): Observable<ProcessFilterRepresentationModel[]> {
+        let key = 'process-filters-' + appName;
+        const filters = JSON.parse(this.storage.getItem(key) || '[]');
+        return new Observable(function(observer) {
+            observer.next(filters);
+            observer.complete();
+        });
+    }
+
+    /**
+     * Adds a new process instance filter
+     * @param filter The new filter to add
+     * @returns Details of process filter just added
+     */
+    addFilter(filter: ProcessFilterRepresentationModel) {
+        const key = 'process-filters-' + filter.query.appName || '0';
+        const storedFilters = JSON.parse(this.storage.getItem(key) || '[]');
+
+        storedFilters.push(filter);
+        this.storage.setItem(key, JSON.stringify(storedFilters));
+    }
+
+    /**
+     * Creates and returns a filter for "All" Process instances.
+     * @param appName Name of the target app
+     * @returns The newly created filter
+     */
+    getAllProcessesFilter(appName: string): ProcessFilterRepresentationModel {
+        return new ProcessFilterRepresentationModel({
+            name: 'All Processes',
+            icon: 'adjust',
+            query: new ProcessQueryModel(
+                {
+                    appName: appName,
+                    sort: 'startDate',
+                    order: 'DESC'
+                }
+            )
+        });
+    }
+
+    /**
+     * Creates and returns a filter for "Running" Process instances.
+     * @param appName Name of the target app
+     * @returns The newly created filter
+     */
+    getRunningProcessesFilter(appName: string): ProcessFilterRepresentationModel {
+        return new ProcessFilterRepresentationModel({
+            name: 'Running Processes',
```

Comment:
can we have a `key`, in this way in case of translation we can rely on it
something like 
```
name: 'PROCESS_CLOUD.FILTERS.RUNNING_PROCESS',
key: 'running-process'
```